alacritty/src/window_context.rs

```diff
@@ -184,8 +189,8 @@ impl WindowContext {
         }
 
         // Live title reload.
-        if !config.window.dynamic_title || old_config.window.title != config.window.title {
-            self.display.window.set_title(&config.window.title);
+        if !config.window.dynamic_title || self.identity.title != config.window.identity.title {
+            self.display.window.set_title(&config.window.identity.title);
         }
```

Comment:
If the CLI option is specified here and the config is changed, should we really change the title? Shouldn't the CLI parameter have precedence?

In that case we might also get away with using `old_config.window.title` so we don't have to store the field on `WindowContext`. Though we might have to store a flag for it?

Reply:
I think we could make title from command line to have precedence here. However to make this work we should check that title was actually passed on CLI(`!dynamic_title && identity.title != old_config.window.identity.title`). But this won't work when you pass title from config on CLI (which isn't something you'd actually do...).

If you want a flag to indicate whether option was from CLI(besides my idea), you need to pass CLI identity to `WindowContext` and then perform override right in `WindowContext::new` (I really don't want to pass one extra argument).

What solution would you prefer?

Reply:
> But this won't work when you pass title from config on CLI

Considering --option the same as changing it in the config is perfectly fine to me. That doesn't need to have precedence.

> I really don't want to pass one extra argument

Same.